core/src/components/segment/test/scrollable/segment.e2e.ts

```diff
@@ -45,3 +45,63 @@ configs().forEach(({ title, screenshot, config }) => {
     });
   });
 });
+
+configs({ modes: ['md'], directions: ['ltr'] }).forEach(({ title, config }) => {
+  test.describe(title('segment: scrollable (functionality)'), () => {
+    test('should scroll active button into view when value is already set', async ({ page }) => {
+      await page.setContent(
+        `
+          <ion-segment scrollable="true" value="8">
+            <ion-segment-button value="1">
+              <ion-label>First</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="2">
+              <ion-label>Second</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="3">
+              <ion-label>Third</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="4">
+              <ion-label>Fourth</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="5">
+              <ion-label>Fifth</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="6">
+              <ion-label>Sixth</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="7">
+              <ion-label>Seventh</ion-label>
+            </ion-segment-button>
+            <ion-segment-button id="activeButton" value="8">
+              <ion-label>Eighth</ion-label>
+            </ion-segment-button>
+            <ion-segment-button value="9">
+              <ion-label>Ninth</ion-label>
+            </ion-segment-button>
+          </ion-segment>
+        `,
+        config
+      );
+
+      const activeButton = page.locator('#activeButton');
+
+      // wait for segment to scroll, which is done in a raf
+      await page.waitForTimeout(500);
+
+      /**
+       * We can't use toBeVisible() here because it returns true
+       * even if the element is outside the viewport.
+       */
+      const box = await activeButton.boundingBox();
```

Comment:
Does `toBeInViewport` work here? https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-in-viewport

Reply:
nice!